src/metabase/lib/join.cljc

```diff
@@ -271,6 +277,117 @@
   [x fields]
   (with-join-fields-method x fields))
 
+(defn- select-home-column
+  [home-cols cond-fields]
+  (let [cond->home (into {}
+                         (keep  (fn [home-col]
+                                  (when-let [cond-field (lib.equality/find-closest-matching-ref
+                                                         (lib.ref/ref home-col)
+                                                         cond-fields)]
+                                    [cond-field home-col])))
+                         home-cols)
+        cond-home-cols (map cond->home cond-fields)]
+    ;; first choice: the leftmost FK or PK in the condition referring to a home column
+    (or (m/find-first (some-fn lib.types.isa/foreign-key? lib.types.isa/primary-key?) cond-home-cols)
+        ;; otherwise the leftmost home column in the condition
+        (first cond-home-cols)
+        ;; otherwise the first FK home column
+        (m/find-first lib.types.isa/foreign-key? home-cols)
+        ;; otherwise the first PK home column
+        (m/find-first lib.types.isa/primary-key? home-cols)
+        ;; otherwise the first home column
+        (first home-cols))))
+
+(defn- strip-id [s]
+  (when (string? s)
+    (str/trim (str/replace s #"(?i) id$" ""))))
+
+(defn- similar-names [name0 name1]
+  (and (string? name0) (string? name1)
+       (let [plural1 (delay (inflections/plural name1))
+             plural0 (delay (inflections/plural name0))]
+         (or (= name0 name1)
+             (= name0 @plural1)
+             (= @plural0 name1)
+             (= @plural0 @plural1)))))
+
+(comment
+  (similar-names "jellow" "yellow")
+  (similar-names "yellow" "yellow")
+  (similar-names "yellow" "yellows")
+  (similar-names "yellow" "yellows")
+  (similar-names "yellows" "yellows")
+  (similar-names "cactus" "cacti")
+  (similar-names "cactus" "cactuses")
+  (similar-names "emacs" "emacsen")
+  nil)
```

Comment:
Maybe you can take all of these test cases and stick them in a unit test for `similar-names`?

Reply:
They are not all valid, I was just experimenting. Removed them.